examples/wakaama/README.md

```diff
@@ -0,0 +1,174 @@
+Eclipse Wakaama Example Client
+==============================
+
+This application is a port of the Eclipse Wakaama [example client
+code](https://github.com/eclipse/wakaama/tree/master/examples/client)
+to RIOT/GNRC.
+
+Usage
+=====
+
+Edit the defines in `lwm2mconfig.h` according to your setup (see the
+comments for what they do).
+
+Build, flash and start the application:
+```
+export BOARD=your_board
+make
+make flash
+make term
+```
+
+(Or Skip `make flash` if you're building for `BOARD=naive`.)
+
+The `term` make target starts a terminal emulator for your board. It
+connects to a default port so you can interact with the shell, usually
+that is `/dev/ttyUSB0`. If your port is named differently, the
+`PORT=/dev/yourport` (not to be confused with the UDP port) variable can
+be used to override this.
+
+
+Available LwM2M Objects
+=======================
+
+At this time, only the mandatory objects for a client are present. The
+*server*, *security* and *access control* objects are taken as is from
```

Comment:
CoAP can be used by not specifying any keys (in this case, when `LWM2M_SERVER_PSK_ID` and `LWM2M_SERVER_PSK_KEY` are set to `NULL`) as that will bypass calling DTLS functions (see e.g. [`connection_handle_packet()`](https://github.com/RIOT-OS/RIOT/pull/8610/files#diff-643b70ce910c75a97d1e160bed99b716R573)). The wakaama example makes it possible to not compile with TinyDTLS so *only* CoAP will work, but with TinyDTLS, both will. Because of that, I did not make the effort to port the non-DTLS version.

Reply:
Ah, now is more clear for me. 

IMHO there is also a factor to consider for RIOT: If DTLS is not going to be used, it should not being compiled in first place. 

This could help us us to reduce the complexity of the testings by removing tinyDTLS stack temporally.  Particularly, if all the lines with DTLS are inside of the files dtlsconnection.* . Otherwise, could be a future work to consider.